search-provider/eks-metadata-provider.c

```diff
@@ -0,0 +1,602 @@
+                                                      G_VARIANT_TYPE_STRING);
+      maybe_add_key_value_pair_from_model_to_variant (model,
+                                                      &builder,
+                                                      "ekn_id",
+                                                      G_VARIANT_TYPE_STRING);
+      maybe_add_key_value_pair_from_model_to_variant (model,
+                                                      &builder,
+                                                      "child_tags",
+                                                      G_VARIANT_TYPE_STRING_ARRAY);
+      maybe_add_key_value_pair_from_model_to_variant (model,
+                                                      &builder,
+                                                      "tags",
+                                                      G_VARIANT_TYPE_STRING_ARRAY);
+
+      g_variant_builder_close (&builder);
+    }
+
+  shards_strv = strv_from_shard_list ((GSList *) shards);
+
+  eks_metadata_complete_query (state->provider->skeleton,
+                                       state->invocation,
+                                       (const char * const *) shards_strv,
+                                       g_variant_builder_end (&builder));
+}
+
+static void
+append_construction_prop_from_string (const char *key,
+                                      const char *str,
+                                      GArray     *values_array,
+                                      GPtrArray  *props_array)
+{
+  guint index = values_array->len;
+  GValue *value = NULL;
+
+  /* Need to be careful to set value *after* the array has been resized */
+  g_array_set_size (values_array, values_array->len + 1);
+  value = &(g_array_index (values_array, GValue, index));
+  g_value_init (value, G_TYPE_STRING);
+  g_value_set_string (value, str);
+  g_ptr_array_add (props_array, g_strdup (key));
+}
+
+typedef gboolean (*VariantToValueTransformFunc) (GVariant  *variant,
+                                                 GValue    *value,
+                                                 gpointer   user_data,
+                                                 GError   **error);
+
+static gboolean
+translate_gvariant_to_gvalue (GVariant  *variant,
+                              GValue    *value,
+                              gpointer   user_data,
+                              GError   **error)
+{
+  g_dbus_gvariant_to_gvalue (variant, value);
+  return TRUE;
+}
+
+G_DEFINE_AUTOPTR_CLEANUP_FUNC (GEnumClass, g_type_class_unref);
+
+static gboolean
+translate_gvariant_to_gvalue_parse_enum (GVariant  *variant,
+                                         GValue    *value,
+                                         gpointer   user_data,
+                                         GError   **error)
+{
+  GType enum_type = (GType) GPOINTER_TO_UINT (user_data);
+  const gchar *str = g_variant_get_string (variant, NULL);
```

Comment:
`const char *` - please check for and replace other instances of `gchar` in new code.

Reply:
Force of habit :(